firefox_ui_tests/functional/locationbar/test_favicon_in_autocomplete.py

```diff
@@ -0,0 +1,87 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import time
+
+from marionette_driver import Wait
+
+from firefox_ui_harness.decorators import skip_under_xvfb
+from firefox_ui_harness import FirefoxTestCase
+
+
+class TestFaviconInAutocomplete(FirefoxTestCase):
+
+    def setUp(self):
+        FirefoxTestCase.setUp(self)
+
+        # Location bar suggests "History"
+        self.prefs.set_pref('browser.urlbar.default.behavior', 1)
+
+        self.test_urls = [self.marionette.absolute_url('layout/mozilla.html')]
+
+        self.test_string = 'mozilla'
+
+        self.autocomplete_results = self.browser.navbar.locationbar.autocomplete_results
+
+        self.places.remove_all_history()
+
+    def tearDown(self):
+        try:
+            self.autocomplete_results.close(force=True)
+        finally:
+            FirefoxTestCase.tearDown(self)
+
+    @skip_under_xvfb
+    def test_favicon_in_autocomplete(self):
+        # Open the test page
+        def load_urls():
+            with self.marionette.using_context('content'):
+                for url in self.test_urls:
+                    self.marionette.navigate(url)
+        self.places.wait_for_visited(self.test_urls, load_urls)
+
+        locationbar = self.browser.navbar.locationbar
+        favicon = self.browser.navbar.locationbar.favicon
+        # this code assigns locationbar_favicon_url an empty string; maybe a sleep will help? no
+        time.sleep(5)
+        locationbar_favicon_url = self.marionette.execute_script("""
+          return arguments[0].getAttribute("src");
+        """, script_args=[favicon])
+        # this code more closely matches mozmill test but throws error:
+        # 'UnboundLocalError: local variable 'wrapped' referenced before assignment'
+        # locationbar_favicon_url = self.marionette.execute_script("""
+        #  return arguments[0].getElement({type:"favicon"}).getNode().getAttribute("src");
+        # """, script_args=[self.browser.navbar.locationbar])
+        #
+        # this code assigns favicon_image a value different from the compared result_icon
+        favicon_image = favicon.value_of_css_property('list-style-image')
+
+        print 'locationbar_favicon_url: %s' % locationbar_favicon_url
+        print 'favicon_image: %s' % favicon_image
+
+        # Clear the location bar, type the test string, check that autocomplete list opens
+        locationbar.clear()
+        locationbar.urlbar.send_keys(self.test_string)
+        self.assertEqual(locationbar.value, self.test_string)
+        Wait(self.marionette).until(lambda _: self.autocomplete_results.is_open)
+
+        result = self.autocomplete_results.visible_results[0]
```

Comment:
This won't work anymore given that the first entry will be the search now due to the landing of the unified urlbar. See more details in bug 1170148.